packages/next/src/cli/next-test.ts

```diff
@@ -0,0 +1,205 @@
+        `Test runner ${configuredTestRunner} is not supported.`
+      )
+  }
+}
+
+async function checkRequiredDeps(
+  baseDir: string,
+  testRunner: SUPPORTED_TEST_RUNNERS
+) {
+  const deps = await hasNecessaryDependencies(
+    baseDir,
+    requiredPackagesByTestRunner[testRunner]
+  )
+  if (deps.missing.length > 0) {
+    await installDependencies(baseDir, deps.missing, true)
+
+    const playwright = spawn(
+      path.join(baseDir, 'node_modules', '@playwright', 'test', 'cli.js'),
+      ['install'],
+      {
+        cwd: baseDir,
+        shell: false,
+        stdio: 'inherit',
+        env: {
+          ...process.env,
+          NODE_ENV: 'test',
+        },
+      }
+    )
+
+    return new Promise((resolve, reject) => {
+      playwright.on('close', (c) => resolve(c))
+      playwright.on('error', (err) => reject(err))
+    })
+  }
+}
+
+async function runPlaywright(
+  baseDir: string,
+  nextConfig: NextConfigComplete,
+  testRunnerArgs: string[]
+) {
+  await checkRequiredDeps(baseDir, 'playwright')
+
+  const playwrightConfigFile = await findUp(
+    ['playwright.config.js', 'playwright.config.ts'],
+    {
+      cwd: baseDir,
+    }
+  )
+
+  if (!playwrightConfigFile) {
+    const { pagesDir, appDir } = findPagesDir(baseDir)
+
+    const { version: typeScriptVersion } = await verifyTypeScriptSetup({
+      dir: baseDir,
+      distDir: nextConfig.distDir,
+      intentDirs: [pagesDir, appDir].filter(Boolean) as string[],
+      typeCheckPreflight: false,
+      tsconfigPath: nextConfig.typescript.tsconfigPath,
+      disableStaticImages: nextConfig.images.disableStaticImages,
+      hasAppDir: !!appDir,
+      hasPagesDir: !!pagesDir,
+    })
+
+    const isUsingTypeScript = !!typeScriptVersion
+
+    const playwrightConfigFilename = isUsingTypeScript
+      ? 'playwright.config.ts'
+      : 'playwright.config.js'
+
+    writeFileSync(
+      path.join(baseDir, playwrightConfigFilename),
+      defaultPlaywrightConfig(isUsingTypeScript)
+    )
+
+    return printAndExit(
+      `Successfully generated ${playwrightConfigFilename}. Create your first test and then run \`next experimental-test\`.`,
+      0
+    )
+  } else {
+    const playwright = spawn(
+      path.join(baseDir, 'node_modules', '@playwright', 'test', 'cli.js'),
+      ['test', ...testRunnerArgs],
+      {
+        cwd: baseDir,
+        shell: false,
+        stdio: 'inherit',
+        env: {
+          ...process.env,
+          NODE_ENV: 'test',
+        },
+      }
+    )
+    return new Promise((resolve, reject) => {
+      playwright.on('close', (c) => resolve(c))
+      playwright.on('error', (err) => reject(err))
+    })
+  }
+}
+
+function defaultPlaywrightConfig(typescript: boolean) {
+  const comment = `/*
+ * Specify any additional Playwright config options here.
+ * They will be deep merged with Next.js' default Playwright config.
+ * You can access the default config by using a function: \`withNext((config) => {})\`
+ */`
+  return typescript
```

Comment:
Do we do the same string concetanation for default tsconfig or eslint config? Reading from a file seems more robust and means existing validation tools in CI check that the config file is actually valid.

Reply:
I mentioned this before, but our TS config fails when we make this standalone because the import path `next/experimental/...` doesn't make sense to it. To avoid that I have to then add the file to ignore paths and all of a sudden we aren't really validating it at all.